### PR TITLE
CMR-4521 Remove UMM-S from UMM-Var

### PR DESCRIPTION
**Pull request by oubiwann**
## First Draft

Super-simple; nothing to it.

That being said: @ygliuvt is this sufficient to address MMT's request? (I _think_ it is ...)

## Second Draft

We decided to do a few more things, and it's not such a simple PR anymore ;-)

 * Update schema
 * Update tests
 * Re-org  EOSDIS Metadata Format Depot / Unified Metadata Model for component-level versioning (different repo)
 * Bring in latest changes from UMM-Var 
 * Refactor schema migration namespaces
 * Add schema migration for this ticket 
 * Update docs that point to UMM-Var 1.0 to point to UMM-Var 1.1 instead

### CODE REVIEW

**Comment by ygliuvt** [comment]
Is this the same as on [Draft|https://git.earthdata.nasa.gov/projects/EMFD/repos/unified-metadata-model/browse/DRAFT/umm-var-json-schema.json]?

**Comment by ygliuvt** [comment]
Need to talk to MMT and Simon as to if we should start versioning UMM-Var now.

**Comment by oubiwann** [comment]
@ygliuvt & @chris-durbin: I've pushed more changes based on the discussions today, complete with refactorings ;-)

File: umm-spec-lib/src/cmr/umm_spec/migration/version/variable.clj [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Variable Migration Implementations
+
+(defmethod interface/migrate-umm-version [:variable "1.0" "1.1"]
+  [context v & _]
+  (dissoc v :Services))
+
+(defmethod interface/migrate-umm-version [:variable "1.1" "1.0"]
+  [context v & _]
+  (migrate-services-down v))
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/variable.clj]: There will not be any services on UMM-Var 1.1 JSON, so this line can be changed to just return `v`. Remove the `migrate-services-down` function above.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  "Contains functions for migrating between versions of UMM schema."
+  (:require
+   [clojure.set :as set]
+   [clojure.string :as string]
+   [cmr.common-app.services.kms-fetcher :as kf]
+   [cmr.common.mime-types :as mt]
+   [cmr.common.util :as util :refer [update-in-each]]
+   [cmr.umm-spec.dif-util :as dif-util]
+   [cmr.umm-spec.location-keywords :as lk]
+   [cmr.umm-spec.migration.collection-progress-migration :as coll-progress-migration]
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj]: Remove unused namespaces.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/interface.clj [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  schemas.")
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Interface for Migrating Between Versions
+
+(defn- dispatch-migrate
+  [_context _concept concept-type source-version dest-version]
+  [concept-type source-version dest-version])
+
+(defmulti migrate-umm-version
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/interface.clj]: I think it is easier to read if you put `dispatch-migrate` function inline. it is also the CMR convention. i.e.

```
(defmulti migrate-umm-version
  ""Returns the given data structure of the indicated concept type and UMM
  version updated to conform to the target UMM schema version."
  (fn [_context _concept concept-type source-version dest-version]
   [concept-type source-version dest-version]))
```

File: umm-spec-lib/src/cmr/umm_spec/migration/version/interface.clj [hunk 4]
```diff
@@ -0,0 +1,22 @@
+(ns cmr.umm-spec.migration.version.interface
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/interface.clj]: I think we can remove this namespace and add the two functions in `cmr.umm-spec.migration.version.core`.

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version/variable.clj [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  (is (nil?
+       (:Services
+        (vm/migrate-umm {} :variable "1.0" "1.1"
+                        {:ServiceTypes ["OPeNDAP"] :Visualizable false :Subsettable false})))))
+
+(deftest migrate-1_1-down-to-1_0
+  (is (nil?
+       (:Services
+        (vm/migrate-umm {} :variable "1.1" "1.0"
+                        {:Services nil})))))
```
Comment (ygliuvt) [inline umm-spec-lib/test/cmr/umm_spec/test/migration/version/variable.clj]: This should just be `{}` as there is no `Services` field in UMM-Var 1.1.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+   [cmr.umm-spec.migration.distance-units-migration :as distance-units-migration]
+   [cmr.umm-spec.migration.geographic-coordinate-units-migration :as geographic-coordinate-units-migration]
+   [cmr.umm-spec.migration.organization-personnel-migration :as op]
+   [cmr.umm-spec.migration.related-url-migration :as related-url]
+   [cmr.umm-spec.migration.spatial-extent-migration :as spatial-extent]
+   [cmr.umm-spec.migration.version.interface :as interface]
+   [cmr.umm-spec.spatial-conversion :as spatial-conversion]
+   [cmr.umm-spec.util :as u]
+   [cmr.umm-spec.versioning :refer [versions current-version]]
+   [cmr.umm-spec.xml-to-umm-mappings.characteristics-data-type-normalization :as char-data-type-normalization]))
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj]: Should require namespaces `cmr.umm-spec.migration.version.collection` and `cmr.umm-spec.migration.version.variable` to support multimethod `interface/migrate-umm-version`. This is why the ci tests are failing.

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version/variable.clj (hunk 5 above)
Comment (oubiwann) [reply umm-spec-lib/test/cmr/umm_spec/test/migration/version/variable.clj]: Whoops -- thanks.

Comment (oubiwann) [reply umm-spec-lib/test/cmr/umm_spec/test/migration/version/variable.clj]: Done.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/variable.clj (hunk 1 above)
Comment (oubiwann) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/variable.clj]: Done.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj (hunk 2 above)
Comment (oubiwann) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj]: Done.

**Comment by oubiwann** [comment]
> Should also update the default UMM-Var version to 1.1 in cmr.ingest.config. i.e.

Done.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj (hunk 6 above)
Comment (oubiwann) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj]: Nice catch.

Comment (oubiwann) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj]: Fixed, pushed, and waiting for build ...

File: umm-spec-lib/src/cmr/umm_spec/migration/version/interface.clj (hunk 3 above)
Comment (oubiwann) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/interface.clj]: Yeah, I can do that. I was a little surprised to see this broken out ... not the usual pattern in the CMR.

Done.

**Comment by oubiwann** [comment]
Hey @ygliuvt -- addressed all feedback & build is green :-) Wanna take another look?

**Comment by oubiwann** [comment]
Thanks!

**Pull request merged**